sdk/search/search-documents/test/public/node/searchClient.spec.ts

```diff
@@ -254,5 +376,154 @@ versionsToTest(serviceVersions, {}, (serviceVersion, onVersions) => {
       const documentCount = await searchClient.getDocumentsCount();
       assert.equal(documentCount, 11);
     });
+    it("search with semantic ranking", async function () {
+      const searchResults = await searchClient.search("luxury", {
+        skip: 0,
+        top: 5,
+        includeTotalCount: true,
+        queryType: "semantic",
+        semanticSearchOptions: { configurationName: "semantic-configuration-name" },
+      });
+      assert.equal(searchResults.count, 1);
+    });
+
+    it("search with answers", async function () {
+      const searchResults = await searchClient.search("What are the most luxurious hotels?", {
+        queryType: "semantic",
+        semanticSearchOptions: {
+          configurationName: "semantic-configuration-name",
+          answers: { answerType: "extractive", count: 3, threshold: 0.7 },
+        },
+        top: 3,
+        select: ["hotelId"],
+      });
+
+      const resultIds = [];
+      for await (const result of searchResults.results) {
+        resultIds.push(result.document.hotelId);
+      }
+      assert.deepEqual(["3", "9", "1"], resultIds);
+    });
+
+    it("search with semantic error handling", async function () {
+      const searchResults = await searchClient.search("luxury", {
+        queryType: "semantic",
+        semanticSearchOptions: {
+          configurationName: "semantic-configuration-name",
+          errorMode: "partial",
+        },
+        select: ["hotelId"],
+      });
+
+      const resultIds = [];
+      for await (const result of searchResults.results) {
+        resultIds.push(result.document.hotelId);
+      }
+      assert.deepEqual(["1"], resultIds);
+    });
+
+    it("search with vector", async function () {
+      // This live test is disabled due to temporary limitations with the new OpenAI service
+      if (isLiveMode()) {
+        this.skip();
+      }
+      const embeddings = await openAIClient.getEmbeddings(
+        env.OPENAI_DEPLOYMENT_NAME ?? "deployment-name",
```

Comment:
should we use `assertEnvironmentVariable` instead and add this variable to the playback map?

Reply:
I wasn't aware that this existed. Thanks!